src/device/device_ledger.cpp

```diff
@@ -1346,6 +1347,75 @@ namespace hw {
         return true;
     }
 
+    void device_ledger::get_transaction_prefix_hash(const cryptonote::transaction_prefix& tx, crypto::hash& h) {
+      AUTO_LOCK_CMD();
+      
+      int pref_length = 0, pref_offset = 0, offset = 0;
+
+      #ifdef DEBUG_HWDEVICE
+      crypto::hash h_x;
+      crypto::hash h_xx;
+      this->controle_device->get_transaction_prefix_hash(tx,h_x);
+      MDEBUG("get_transaction_prefix_hash [[IN]] h_x/1 "<<h_x);
+      #endif
+    
+      std::ostringstream s_x;
+      binary_archive<true> a_x(s_x);
+      ::serialization::serialize(a_x, const_cast<cryptonote::transaction_prefix&>(tx));
```

Comment:
Return code not checked

Reply:
I juste copy/paste the orignal code. I will add check

Reply:
the function return void

Reply:
That's odd. Can you point to it ? The ones I see return bool but I could be missing some overrides.

Reply:
I found this one https://github.com/monero-project/monero/blob/release-v0.15/src/cryptonote_basic/cryptonote_boost_serialization.h#L153

But I'm sure at all about if it is the right one or not

Reply:
These are for the boost serialization, used for files (ie, saving the wallet cache). Wire serialization is not that, since boost serialization is dependent on boost version and endianness. You want src/serialization/serialization.h

Reply:
thanks for the pointer. I will check

Reply:
I added a CHECK_AND_ASSERT_THROW_MES